### PR TITLE
Create CNCF-DarshanKumar-build_imgext.md

### PR DESCRIPTION
**Pull request by imohitpatel**
Add darshan Kumar (@itsdarshankumar) gsoc proposal of 2023  to Folder Gsoc2023 -> CNCF-DarshanKumar-build_imgext. ->CNCF-DarshanKumar-build_imgext.md

[x] Followed contribution guidelines

### CODE REVIEW

**Comment by imohitpatel** [comment]
Please @saketkc merge
 asked to dranshan for permission

**Comment by saketkc** [comment]
Thank your very much for the PR! Since the original proposal is of @itsdarshankumar, I would need his permission to go ahead before merging.

File: GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+
+**Current Year:** 2nd year (May 2025 expected graduation)
+
+
+
+**Degree:** Bachelor of Technology (B.Tech)
+
+
+
+16![](media/Aspose.Words.a9fed3b9-8a2f-41f6-a3ed-332fa7b604b3.001.png)
```
Comment (itsdarshankumar) [inline GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md:62]: is it relevant?

File: GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+
+### What are your expectations from us during and after the successful completion of the program?
+
+
+
+I look forward to bonding with the community and contributing valuable contributions to the project. I would be grateful to join the family of Cloud Native Buildpacks and CNCF. I am always open to essential discussions over ideas and their implementations. I would love to receive feedback and suggestions on my contributions, as it willhelp improve my code quality and help me become a better developer in the long run.
+
+
+
+As Imentioned above, your community’s projects drive me to make consistent contributions to the work it is doing. The GSoC project allows me to work on the project as a whole, and after its successful completion, Iwould be keen to take up more important issues and continue my open-source contribution through Cloud Native Buildpacks and CNCF.
```
Comment (itsdarshankumar) [inline GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md:110]: `Imentioned` -> I mentioned

File: GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+
+
+
+- `**pack**` already has access to the host’s daemon (**Docker**),so it can apply the generated dockerfiles (**build.Dockerfile** and **run.Dockerfile**) directly, saving the extended build or runtime base image in the host’s daemon itself. Thus it willnot need to use the extender phase of the lifecycle. Further,having this feature will allow us to drop the requirement that the builder is published to a registry in case of extension and **Docker** daemon builds.
+
+- On the other hand, the **docker** daemon can cache the layers of an image and only rebuild the layers that have changed, which can make builds faster,thus significantly reducing the runtime for an image extension. In some cases, we would need two separate `**extend**` operations -one for the build image and the other for the run image. They can happen in parallel by extending the build image and then proceeding to the `**build**` phase. Then after extending the run image and the build phase has been completed, we can proceed to the export phase, thus saving a lot of time.
+
+
+
+**ImplementationDetails**
```
Comment (itsdarshankumar) [inline GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md:280]: Grammatical error

File: GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+
+
+
+Redefiningthe`Run`Function
```
Comment (itsdarshankumar) [inline GSoC-2023/Accepted/CNCF-DarshanKumar-build_imgext./CNCF-DarshanKumar-build_imgext.md:284]: Grammatical errors

**Comment by itsdarshankumar** [comment]
Hello @saketkc @imohitpatel 
You can merge this as far as this will be used for free to promote open-source activities.

**Pull request closed without merging**